src/main.rs

```diff
@@ -24,7 +24,11 @@ pub use catchup::Catchup;
 async fn main() -> anyhow::Result<()> {
     // Configuring logging
     let fmt_layer = tracing_subscriber::fmt::layer().with_target(true);
-    let filter_layer = EnvFilter::try_from_default_env().or_else(|_| EnvFilter::try_new("info"))?;
+    let filter_layer = EnvFilter::try_from_default_env()
+        .or_else(|_| EnvFilter::try_new("info"))?
+        // Force disabling of r1cs log messages, otherwise the `ark-groth16` crate
+        // can use a massive (>16GB) amount of memory generating unused trace statements.
+        .add_directive("r1cs=off".parse()?);
```

Comment:
Unrelated to compat changes, but had this commit lying around, and it should indeed be included, see https://github.com/penumbra-zone/penumbra/issues/2977